### DISCUSSION TITLE
Some more xxxx_FIXED ops would be nice

### DISCUSSION THREAD

**Discussion by YuriMyakotin** [axboe/liburing#679] [Ideas]
Currently IO_URING have only 2 i/o operations with fixed buffers - IORING_OP_WRITE_FIXED and IORING_OP_READ_FIXED.

Is there any way to add in future versions new ops - IORING_OP_SEND_FIXED/IORING_OP_RECV_FIXED (same as write/read, but with  flags support, especially MSG_WAITALL for receive)?
And, ofc, UDP versions - IORING_OP_SENDTO_FIXED and IORING_OP_RECVFROM_FIXED?

**Comment by isilence** [comment]
> Currently IO_URING have only 2 i/o operations with fixed buffers -
> IORING_OP_WRITE_FIXED and IORING_OP_READ_FIXED.
>
> Is there any way to add in future versions new ops -
> IORING_OP_SEND_FIXED/IORING_OP_RECV_FIXED (same as write/read, but with
> flags support, especially MSG_WAITALL for receive)?
> And, ofc, UDP versions - IORING_OP_SENDTO_FIXED and
> IORING_OP_RECVFROM_FIXED?
>

Normal send/recc would have nearly zero performance benefit because of how
it internally works. New zerocopy send opcode though supports registered
buffers (see SENDRECV iouring flags). I was thinking to add support just
for consistency but later and only if it pans out well

>
